src/services/codefixes/fixStrictClassInitialization.ts

```diff
@@ -0,0 +1,141 @@
+/* @internal */
+namespace ts.codefix {
+    const fixIdAddDefiniteAssignmentAssertions = "addMissingPropertyDefiniteAssignmentAssertions";
+    const fixIdAddUndefinedType = "addMissingPropertyUndefinedType";
+    const fixIdAddInitializer = "addMissingPropertyInitializer";
+    const errorCodes = [Diagnostics.Property_0_has_no_initializer_and_is_not_definitely_assigned_in_the_constructor.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions: (context) => {
+            const propertyDeclaration = getPropertyDeclaration(context.sourceFile, context.span.start);
+            if (!propertyDeclaration) return undefined;
+
+            const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+            return [
+                getActionsForAddMissingDefiniteAssignmentAssertion(context, propertyDeclaration, newLineCharacter),
+                getActionsForAddMissingUndefinedType(context, propertyDeclaration),
+                getActionsForAddMissingInitializer(context, propertyDeclaration, newLineCharacter)
+            ].filter(x => !!x);
+        },
+        fixIds: [fixIdAddDefiniteAssignmentAssertions, fixIdAddUndefinedType, fixIdAddInitializer],
+        getAllCodeActions: context => {
+            const seenNames = createMap<true>();

```

Comment:
Are you sure we need this? I don't think we would have two of these errors for the same property. But we might get this error for two separate properties that happen to share a name, which would result in only one fix here.